src/cljs/harja/views/urakka/pot_yhteinen.cljs

```diff
@@ -414,27 +447,33 @@
                ::lomake/col-luokka "col-xs-12"})]
            perustiedot-nyt]]]))))
 
+
 (defn kasittely [e! {:keys [perustiedot] :as app} urakka lukittu?
                  muokkaa! validoinnit huomautukset]
   (let [{:keys [tila asiatarkastus versio]} perustiedot
         nayta-kasittelyosiot? (#{:valmis :lukittu} tila)
         asiatarkastus-sis-tietoja? (some #(some? (val %))
                                          (lomake/ilman-lomaketietoja asiatarkastus))]
-
     (fn [e! {:keys [perustiedot] :as app} urakka lukittu?
          muokkaa! validoinnit huomautukset]
-      [:div.kasittelyosio
-       [:h5 "Käsittely"]
-       (if-not nayta-kasittelyosiot?
-         [kentat/tee-kentta {:tyyppi :checkbox
-                             :teksti "Merkitse valmiiksi tarkistusta varten"}
-          (r/wrap (get-in app [:perustiedot :valmis-kasiteltavaksi])
-                  (fn [uusi]
-                    (e! (paallystys/->AsetaKasiteltavaksi uusi))))]
-         [:span.asiatarkastus-checkbox
-          (when-not (or (= :lukittu tila) asiatarkastus-sis-tietoja?)
-            [kentat/tee-kentta {:tyyppi :checkbox
-                                :teksti "Kaksi tarkastajaa (asiatarkastus erikseen)"} tee-asiatarkastus?])
-          [:div.pot-kasittely
-           [kasittely-asiatarkastus urakka perustiedot lukittu? muokkaa! huomautukset asiatarkastus-sis-tietoja?]
-           [kasittely-tekninen-osa urakka perustiedot lukittu? muokkaa! huomautukset]]])])))
+      (let [kokonaishinta (toteuman-kokonaishinta-hae-fn perustiedot)
+            hinta-puuttuu? (not (and kokonaishinta (> kokonaishinta 0)))]
+        [:div.kasittelyosio
+         (when hinta-puuttuu?
+           [hintatiedot-puuttuvat-komp e! true])
+         (if-not nayta-kasittelyosiot?
+           [kentat/tee-kentta {:tyyppi :checkbox
+                               :teksti "Merkitse valmiiksi tarkistusta varten"
+                               :disabled? hinta-puuttuu?}
```

Comment:
Aina hieman hankala lukea koodista näitä varmistuksia, mutta tässä on nähtävästi homman pihvi, että ei anneta tallentaa ennen kokonaishintaa. Eli minusta asia kunnossa.

Reply:
Juuri näin, tässä estetään lomakkeen merkitseminen valmiiksi, eli käsittely. Lomakkeen tietoja voi kyllä "välitallentaa" myös ilman kustannustietoa eli sikäli ei aiheudu käyttäjille ylimääräistä harmia.